openmc/deplete/abc.py

```diff
@@ -701,6 +712,33 @@ def solver(self, func):
 
         self._solver = func
 
+    @property
+    def reactivity_control(self):
+        return self._reactivity_control
+
+    @reactivity_control.setter
+    def reactivity_control(self, reactivity_control):
+        if isinstance(reactivity_control, CellReactivityController):
+            if not isinstance(reactivity_control.cell, Cell):
+                raise TypeError(
+                    "Reactivity control attribute must be Cell "
+                    "not {}".format(type(reactivity_control.cell)))
+        elif isinstance(reactivity_control, MaterialReactivityController):
+            if not isinstance(reactivity_control.material, Material):
+                raise TypeError(
+                    "Reactivity control attribute must be Material "
+                    "not {}".format(type(reactivity_control.material)))
+        else:
+            raise TypeError(
+                "Reactivity Control must either be "
+                "CellReactivityController, or MaterialReactivityController, "
+                "not {}".format(type(reactivity_control)))
+
+        check_value('attribute', reactivity_control.attrib_name,
+                    ('translation', 'rotation', 'temperature', 'refuel'))
```

Comment:
Why only these four control mechanisms? What about the interplay between the `Operator` and the `ReactivityController` requires only these four mechanisms?

For light water reactors, adding and changing soluble boron concentration might be akin to "refueling" but we're not adding fuel there so someone could set `attrib_name = "dillution"` (for example).